src/videoTrackingHTML5.js

```diff
@@ -0,0 +1,248 @@
+/**
+ * Video tracking for HTML5 videos
+ * 
+ * This script analyses the interactions with <video> elements.
+ */
+//4Patryk: I'm not sure whether to put "subscribe" here at all?
+ export default (subscribe, {eventCategoryLabel,
+                            videoTitleAttribute,
+                            trackingAccuracy,
+                            trackThresholds,
+                            thresholdsToTrack,
+                            trackTimestampAsDimension,
+                            dimensionIdForTimestamps,
+                            trackVolumeAsDimension,
+                            dimensionIdForVolume
+    }) => {
+    
+    const percentageThresholds = thresholdsToTrack.split(",").map(x=>+x);
+    let trackedThresholds = {};
+    let lastTrackedVolume = 100;
+
+    const getVideoName = (video) => {
+        var videoName = video.getAttribute(videoTitleAttribute);
+        if (videoName == null) {
+            if (video.hasAttribute("src")){
+                videoName = video.getAttribute("src").split("/").slice(-1).pop();
+            }
+            else {
+                videoName = video.querySelector("source").getAttribute("src").split("/").slice(-1).pop();
+            }
+        }
+        return videoName;
+    };
+
+    const trackEvent = (eventData) => {
+            if (trackingAccuracy < 0 || trackingAccuracy > 3){
+                trackingAccuracy = 0;
+            }
+            eventData.eventTimestampFixed = eventData.eventTimestamp.toFixed(trackingAccuracy);
+            eventData.eventCategoryLabel = eventCategoryLabel;
+            let eventArray = [
+                'trackEvent',
+                eventData.eventCategoryLabel,
+                eventData.eventType,
+                eventData.videoTitle,
+                eventData.eventTimestampFixed
+            ];
+            if (trackTimestampAsDimension || trackVolumeAsDimension){
+                let dimensionsObject = {};
+                if (trackTimestampAsDimension){
+                    dimensionsObject["dimension"+dimensionIdForTimestamps] = eventData.eventTimestamp;
+                }
+                if (trackVolumeAsDimension){
+                    dimensionsObject["dimension"+dimensionIdForVolume] = encodeURIComponent((eventData.currentVolume*100).toFixed(0)+"%");
+                }
+                eventArray.push(dimensionsObject);
+            }
+            window._paq.push(eventArray);
+    };
+
+    const processPlayMediaEvent = (e) => {
+        var currentPlayTime = e.target.currentTime;
+        var videoTitle = getVideoName(e.target);
+        e.target.currentPlayTime = e.target.currentTime;
+        var isUserSeeking = e.target.seeking;
+        let eventData = {
+            videoTitle: videoTitle,
+            eventTimestamp: currentPlayTime,
+            currentVolume: e.target.volume
+        };
+        if (isUserSeeking == false && e.target.hasPlayed == true && e.target.hasPaused == true) {
```

Comment:
We can omit that assertions and below, eg.:
```suggestion
        if (!isUserSeeking && hasPlayed && hasPaused == true) {
```

Reply:
done